### ISSUE TITLE
General Accessibility Audit

### ISSUE THREAD

**Issue by amandadupell** [grommet/hpe-design-system#1821]
# Purpose of this issue

The purpose of this issue is to identify keyboard/ARIA accessibility improvements that can be made to the Design System UI/UX.

## How were these improvements identified?

Most of the issues came from looking at the Design System pages and taking note of the experience when using a screen reader.

### Actions
- [ ] most onClick handlers should include an announcement that an action has been performed or that there was a change in UI/UX, etc.

For example, when clicking on a card, it should be announced to the user that they are being directed to a new page that card links to. If the user clicks on the Button card, the screen reader should relay something like "opening Button page", etc.

<img width="250" alt="card" src="https://user-images.githubusercontent.com/38971752/131908464-1bc01807-444d-4bd0-b4db-2e6e4fa581f6.png">

### Sections
- [ ] if there is related content under a section, the user should get some sort of indication that the content is related to that section and to the other content within that section

For example, when entering the Philosophy section, the screen reader could relay that the user is entering a section with x amount of cards. When the user accesses a card under the Philosophy section, it should be clear that the card relates to that section.

<img width="500" alt="section" src="https://user-images.githubusercontent.com/38971752/131908833-3531addb-3b74-424a-aae4-74f90abf6462.png">

### Component and Code Examples
- [ ] at the page-level for components, there should be an understanding of how the content of the page relates to the coding examples
- [ ] there should be a clear distinction between the content of the page and the examples
- [ ] there should be a clear distinction between the "playground" where the component example lives and the footer of the example where the corresponding links can be found

For example, for this checkbox, the example is directly related to creating a CheckBox with a description. When the user enters the "playground", it should be relayed to the user that this example is related to the above content. When the user exits the "playground" to access the links in the footer, it should be relayed that they have exited the example and are now interacting with links related to the example.

<img width="500" alt="code example" src="https://user-images.githubusercontent.com/38971752/131909956-5009dcd4-b77b-4454-b981-c13e787567b5.png">
